mbed_targets/_internal/target_attribute_hierarchy_parsers/accumulating_attribute_parser.py

```diff
@@ -13,18 +13,31 @@
 from typing import Dict, List, Any, Deque
 
 ACCUMULATING_ATTRIBUTES = ["extra_labels", "macros", "device_has", "features", "components"]
-ALL_ACCUMULATING_ATTRIBUTES = ACCUMULATING_ATTRIBUTES + [
-    "extra_labels_remove",
-    "extra_labels_add",
-    "macros_remove",
-    "macros_add",
-    "device_has_remove",
-    "device_has_add",
-    "features_remove",
-    "features_add",
-    "components_remove",
-    "components_add",
-]
+ACCUMULATING_ATTRIBUTES_MODIFIERS = {"add": "_add", "remove": "_remove"}
```

Comment:
Is there a reason I'm missing why this is a dict instead of a list? 🤔 
I.e.: `MODIFIERS = ["add", "remove"]`?

Reply:
Only so I don't have to do a separate check that the modifying action I pass to _modifying_attribute_name is in the list, since its a dictionary it will just tell me its not there. But if this isn't a good reason I am happy to change it.

Reply:
You can achieve the same result with a list - `"add" in ["add", "remove"]`. 
I think this dictionary would make more sense if `key` was substantially different to `suffix`. Am I missing something? 🤔

Reply:
No thats fair. I'm redoing it :)

Reply:
45ed7ec